src/dash-table/dash/DataTable.js

```diff
@@ -1,44 +1,47 @@
-import * as R from 'ramda';
-import React, { Component } from 'react';
+import React, { Component, lazy, Suspense } from 'react';
 import PropTypes from 'prop-types';
+import LazyLoader from 'dash-table/LazyLoader';
+import LazyFactory from './LazyFactory';
+
+const isLazyComponentReady = (target, promise) => {
+    let resolve;
+    const isReady = new Promise(r => {
+        resolve = r;
+    });
+
+    const state = {
+        isReady,
+        get: lazy(() => {
+            return Promise.resolve(promise()).then(res => {
+                setTimeout(async () => {
+                    await resolve(true);
+                    state.isReady = true;
+                }, 0);
+
+                return res;
+            });
+        })
+    };
```

Comment:
This code is currently duplicated between dcc and the table, need to share this. At the moment, the most natural way seems to expose it as part of `window.dash_renderer`